### PR TITLE
BUG: support null properties in GeoJSON

### PR DESCRIPTION
**Pull request by clausmichele**
Fixes https://github.com/geopandas/geopandas/issues/2242

It allows to have `'properties':null` in a geoJSON and still allow to use it with GeoDataFrame.from_features.
The geoJSON spec allows to use null for properties: https://geojson.org/geojson-spec.html https://datatracker.ietf.org/doc/html/rfc7946

### CODE REVIEW

File: geopandas/tests/test_geodataframe.py [hunk 1]
```diff
@@ -594,15 +594,15 @@ def test_from_features_unaligned_properties(self):
         p3 = Point(3, 3)
         f3 = {
             "type": "Feature",
-            "properties": {"a": 2},
+            "properties": None,
             "geometry": p3.__geo_interface__,
```
Comment (martinfleis) [inline geopandas/tests/test_geodataframe.py:598]: Can you keep this as it is (including the change below) and add a new test using the snippet with `null` you posted in #2242?

**Review by martinfleis — COMMENTED** [review]
Thanks for the report and a quick fix!

I have just one request regarding tests (see the note in the code).

File: geopandas/tests/test_geodataframe.py (hunk 1 above)
Comment (clausmichele) [reply geopandas/tests/test_geodataframe.py:598]: I've added a new test, let me know if it's fine!
Cheers

**Review by martinfleis — APPROVED** [review]
Thanks!

**Review by jorisvandenbossche — APPROVED** [review]
Sorry for the slow follow-up, but this looks good. Thanks a lot for your contribution!
